### PR TITLE
Various bugfixes

### PR DESCRIPTION
**Pull request by brianhuffman**
Having a bunch of separate parallel pull requests was going to get unwieldy, and merging was going to make a big mess of the commit history, so I'm combining a few separate PRs into this new one (#714, #715, #716).

This PR contains fixes for #691, #698, and #707.
EDIT: Also #701.

### CODE REVIEW

**Comment by kquick** [comment]
This seems to have broken the kyber proofs, possibly because of the argument type checking: http://fryingpan.dev.galois.com/hydra/build/3412946/log

**Comment by brianhuffman** [comment]
What I think you meant to say is, this has *revealed* that the kyber proofs were *already* broken :)

**Comment by brianhuffman** [comment]
But seriously, where can I find these kyber proofs so I can test them and fix them?

**Comment by chameco** [comment]
I've fixed this problem- a Cryptol literal being passed as an argument was annotated as `[32]` where `[64]` was expected.

The Kyber proof is available on the external GitLab. I think you should have access there @brianhuffman.

File: src/SAWScript/Crucible/Common/Override.hs [hunk 1]
```diff
@@ -180,10 +180,10 @@ ppOverrideFailureReason ::
   ) => OverrideFailureReason ext -> PP.Doc
 ppOverrideFailureReason rsn = case rsn of
   AmbiguousPointsTos pts ->
-    PP.text "ambiguous collection of points-to assertions" PP.<$$>
+    PP.text "LHS of points-to assertion(s) not reachable via points-tos from inputs/outputs:" PP.<$$>
```
Comment (andreistefanescu) [inline src/SAWScript/Crucible/Common/Override.hs:183]: "not reachable" -> "not uniquely determined"?

File: src/SAWScript/Crucible/Common/Override.hs [hunk 2]
```diff
@@ -180,10 +180,10 @@ ppOverrideFailureReason ::
...
     (PP.indent 2 $ PP.vcat (map PP.pretty pts))
   AmbiguousVars vs ->
-    PP.text "ambiguous collection of variables" PP.<$$>
+    PP.text "Fresh variable(s) not reachable via points-tos from function inputs/outputs:" PP.<$$>
```
Comment (andreistefanescu) [inline src/SAWScript/Crucible/Common/Override.hs:186]: "not reachable" -> "not uniquely determined"?

**Comment by brianhuffman** [comment]
@atomb you might like to try looking at each commit separately, instead of the overall diff. The reformatting was separated into its own commit.

File: src/SAWScript/Crucible/Common/Override.hs (hunk 1 above)
Comment (brianhuffman) [reply src/SAWScript/Crucible/Common/Override.hs:183]: Is there a situation where a variable could be *determined*, but not *uniquely* so?

**Comment by atomb** [comment]
Ah, right, of course. Looking at the individual commits (but ignoring the reformatting one), I think it all looks good.

**Pull request merged**